compiler/src/iree/compiler/Codegen/Common/EraseHALDescriptorTypeFromMemRef.cpp

```diff
@@ -56,6 +58,35 @@ struct MemRefTypeConverter final : public TypeConverter {
   }
 };
 
+struct GPUMemRefTypeConverter final : public TypeConverter {
+  GPUMemRefTypeConverter() {
+    // Pass through for all other types.
+    addConversion([](Type type) { return type; });
+
+    addConversion([](BaseMemRefType memRefType) -> std::optional<Type> {
+      // Expect #hal.descriptor_type memory spaces.
+      Attribute spaceAttr = memRefType.getMemorySpace();
+      if (!spaceAttr)
+        return std::nullopt;
+      auto dtAttr = llvm::dyn_cast<IREE::HAL::DescriptorTypeAttr>(spaceAttr);
```

Comment:
Do we not care about the HAL descriptor type's space address at all and just always convert to global space?

Reply:
Good point. According to Lei, Uniform Buffers are probably constant so we can lower to that instead.

Reply:
After more investigation there is no memory space upstream for constant so I'll do that as a follow up and land this as is for now.